cosmwasm/utils.js

```diff
@@ -838,6 +852,45 @@ const submitProposal = async (client, wallet, config, options, content) => {
     return events.find(({ type }) => type === 'submit_proposal').attributes.find(({ key }) => key === 'proposal_id').value;
 };
 
+function getCosmWasmUrl(contractName, contractVersion) {
+    if (!SUPPORTED_COSMWASM_CONTRACTS.has(contractName)) {
+        throw new Error(`Unsupported contract ${contractName} for versioned deployment`);
+    }
+
+    const pathName = pascalToKebab(contractName);
+    const fileName = pascalToSnake(contractName);
+
+    const versionRegex = /^v\d+\.\d+\.\d+$/;
+    const commitRegex = /^[a-f0-9]{7,}$/;
+
+    let contractPath;
+
+    if (versionRegex.test(contractVersion)) {
+        // remove leading v
+        const semanticVersion = contractVersion.slice(1);      
+        contractPath = `${AXELAR_R2_BASE_URL}/releases/cosmwasm/${pathName}/${semanticVersion}/${fileName}.wasm`;
+    } else if (commitRegex.test(contractVersion)) {
+        contractPath = `${AXELAR_R2_BASE_URL}/pre-releases/cosmwasm/${contractVersion}/${fileName}.wasm`;
+    } else {
+        throw new Error(`Invalid contractVersion format: ${contractVersion}. Must be a semantic version (including prefix v) or a commit hash`);
+    }
+
+    return contractPath;
+}
+
+async function getWasmPath(options, contractName) {
+    if (options.artifactPath) {
+        return `${options.artifactPath}/${pascalToSnake(contractName)}.wasm`;
+    }
+
+    if (options.version) {
+        const url = getCosmWasmUrl(contractName, options.version);
+        return await downloadWasmFile(url , contractName, options.version);
+    }
+
+    throw new Error('Either --artifactPath or --version must be provided');
+}
+
```

Comment:
could we try moving this to a Commander hook and store the result into a third option or something like that? I think it could be good to move "options" logic earlier in the execution, specially since it's used by both `deploy-contract` and `submit-proposal` scripts and this can avoid some code duplication. Maybe ask @milapsheth because I think I saw him commenting somewhere else about not changing the value of an option